### DISCUSSION TITLE
Update Field limitted choice

### DISCUSSION THREAD

**Discussion by mtspl** [Budibase/budibase#7021] [Help]
Hi, can someone please advice on why i have only two fileds from a givent form/data source available in the update field action ? i would like to change WriteStatus field, but this field is not available :( 

![image](https://user-images.githubusercontent.com/5992234/182153565-7a108fe5-74f6-4e3d-954f-6c5f9e4b4a27.png)

**Comment by R2bEEaton** [comment]
Hey! I'll refer you to #7033 which I just answered and is very similar to your question.

The key thing here is that "Update Field" refers to Form fields, while "Save Row" refers to actual table rows. If you want to change the WriteStatus of a row, you need to use "Save Row."

Good luck!

**Reply by melohagan** [reply]
@mtspl I also don't see the 'WriteStatus' field in your screenshot.

**Reply by mtspl** [reply]
on the screenshot you can see checkbox in a column on the far right. This is a form field - writestatus. an i would like to update chosen record value of that field. as i understood R2bEEaton comment, update field should allow to change this form field value. but that is not the case :(

**Reply by melohagan** [reply]
Update Field Value should allow you to set WriteStatus if it is a checkbox form field. Could you share a screenshot of your component tree please?
The WriteStatus checkbox should be nested within the AvailableCardListForm.

**Reply by mtspl** [reply]
So i have a table in which, all relevant fields are displayed. see the image below. Whats interesting, is that neither CardNo nor RAW, is present as a separate control/component. All of them, are within the table (not table block). 

![image](https://user-images.githubusercontent.com/5992234/183391246-7e9829ca-65d2-4810-b247-11de6b8eaa1f.png)

**Reply by melohagan** [reply]
The WriteStatus is a table field, it's not a form field. The Update Field Value action only works for form fields.

**Reply by mtspl** [reply]
Ok, i understood. So what would be the best approach to show a record list with all the presented in the table fields ?

**Reply by melohagan** [reply]
Apologies @mtspl I'm not quite understanding this:
> record list with all the presented in the table fields

**Reply by mtspl** [reply]
sorry. let me phrase this the other way. Basically i would like to show the same info as in table, but i a ways that would make possible using the update field action.

**Reply by melohagan** [reply]
Thank you, I believe I understand now! The thing here is that you have many rows.
So you could instead of using a table have a Repeater with a nested form. That would probably be a bit messy though, so what I recommend is having a separate screen for editing each row using `:id` url variable to identify the row you're on: https://docs.budibase.com/docs/url-parameters#link-navigation

**Reply by mtspl** [reply]
@melohagan many thanks for your suggestion. this table is not intended to be used as an row edit, but rather as a list of available cards. once chosen by user, it is written on the medium, and this fact should be noted as a writestatus change. but this is only one value that would be changed in the record. anyways, ill try a Repeater approach. thanks again :)
